node/src/components/consensus/era_supervisor.rs

```diff
@@ -219,7 +219,9 @@ where
 
     fn booking_block_height(&self, era_id: EraId) -> u64 {
         // The booking block for era N is the last block of era N - AUCTION_DELAY - 1
-        // To find it, we get the start height of era N - AUCTION_DELAY and subtract 1
+        // To find it, we get the start height of era N - AUCTION_DELAY and subtract 1.
+        // We make sure not to go below the last upgrade activation point, because we will not have
+        // any eras from before that.
```

Comment:
We could make that a doc comment on the method. It's not just an implementation detail but also something the caller should be aware of.